cmd/activator/main.go

```diff
@@ -339,3 +339,14 @@ func flush(logger *zap.SugaredLogger) {
 	os.Stderr.Sync()
 	metrics.FlushExporter()
 }
+
+func wrapActivatorHandlerWithFullDuplex(h http.Handler) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		revEnableHTTP1FullDuplex := activatorhandler.RevAnnotations(r.Context(), apiconfig.AllowHTTPFullDuplexFeatureKey) == "Enabled"
+		if revEnableHTTP1FullDuplex {
+			rc := http.NewResponseController(w)
+			_ = rc.EnableFullDuplex()
```

Comment:
Can we log the error?

Reply:
The only error returned is if the feature is supported or not:
```
// EnableFullDuplex indicates that the request handler will interleave reads from Request.Body
// with writes to the ResponseWriter.
//
// For HTTP/1 requests, the Go HTTP server by default consumes any unread portion of
// the request body before beginning to write the response, preventing handlers from
// concurrently reading from the request and writing the response.
// Calling EnableFullDuplex disables this behavior and permits handlers to continue to read
// from the request while concurrently writing the response.
//
// For HTTP/2 requests, the Go HTTP server always permits concurrent reads and responses.
func (c *ResponseController) EnableFullDuplex() error {
	rw := c.rw
	for {
		switch t := rw.(type) {
		case interface{ EnableFullDuplex() error }:
			return t.EnableFullDuplex()
		case rwUnwrapper:
			rw = t.Unwrap()
		default:
			return errNotSupported()
		}
	}
}
```
		case rwUnwrapper:
			rw = t.Unwrap()
```
This part is used in tests it seems.
Since this is internal and we set `rc := http.NewResponseController(w)`  that should not return an error as the writer passed always supports that operation in 1.21. However, I could report the potential error.

Reply:
I will log the error as it might be the case that a handler does not support the operation.